internal/controller/services/monitoring/monitoring_controller_actions.go

```diff
@@ -95,43 +97,108 @@ func updatePrometheusConfigMap(ctx context.Context, rr *odhtypes.ReconciliationR
-	ok, _, _ := checkDSCI(ctx, rr)
-	if !ok {
-		return nil
-	}
-
 	monitoring, ok := rr.Instance.(*serviceApi.Monitoring)
 	if !ok {
 		return errors.New("instance is not of type *services.Monitoring")
 	}
 
+	// Only create monitoring stack if monitoring is managed and has metrics config
+	if monitoring.Spec.ManagementState != operatorv1.Managed || monitoring.Spec.Metrics == nil {
+		return nil
+	}
+
 	msExists, _ := cluster.HasCRD(ctx, rr.Client, gvk.MonitoringStack)
 	if !msExists {
 		return errors.New("MonitoringStack CRD not found")
 	}
-	if monitoring.Spec.Metrics != nil {
-		template := []odhtypes.TemplateInfo{
-			{
-				FS:   resourcesFS,
-				Path: MonitoringStackTemplate,
-			},
-		}
-		rr.Templates = append(rr.Templates, template...)
+
+	template := []odhtypes.TemplateInfo{
+		{
+			FS:   resourcesFS,
+			Path: MonitoringStackTemplate,
+		},
 	}
+	rr.Templates = append(rr.Templates, template...)
 
 	return nil
 }
 
-func checkDSCI(ctx context.Context, rr *odhtypes.ReconciliationRequest) (bool, *dsciv1.DSCInitialization, error) {
-	dsci, err := cluster.GetDSCI(ctx, rr.Client)
-	// DSCI not found
-	if err != nil && k8serr.IsNotFound(err) {
-		return false, nil, nil
+// handleInstrumentationCR manages OpenTelemetry Instrumentation CRs using server-side apply.
+func handleInstrumentationCR(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	monitoring, ok := rr.Instance.(*serviceApi.Monitoring)
+	if !ok {
+		return errors.New("instance is not of type *serviceApi.Monitoring")
 	}
-	// DSCI found but error
+
+	dsci, err := cluster.GetDSCI(ctx, rr.Client)
 	if err != nil {
-		return false, nil, fmt.Errorf("failed to get DataScienceClusterInitialization instance: %w", err)
+		if k8serr.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to get DSCInitialization instance: %w", err)
 	}
 
-	return true, dsci, nil
+	instrumentationName := "opendatahub-instrumentation"
+	instrumentationNamespace := monitoring.Spec.Namespace
+
+	existingInstrumentation := &unstructured.Unstructured{}
+	existingInstrumentation.SetGroupVersionKind(gvk.Instrumentation)
+	existingInstrumentation.SetName(instrumentationName)
+	existingInstrumentation.SetNamespace(instrumentationNamespace)
+
+	existingErr := rr.Client.Get(ctx, client.ObjectKeyFromObject(existingInstrumentation), existingInstrumentation)
+	instrumentationExists := existingErr == nil
+
+	switch dsci.Spec.Monitoring.ManagementState {
+	case operatorv1.Managed:
+		// Only create instrumentation CR if monitoring is managed and has traces config
+		if monitoring.Spec.ManagementState != operatorv1.Managed || monitoring.Spec.Traces == nil {
+			// If traces are not configured but instrumentation exists, remove it
+			if instrumentationExists {
+				if err := rr.Client.Delete(ctx, existingInstrumentation); err != nil && !k8serr.IsNotFound(err) {
+					return fmt.Errorf("failed to delete instrumentation CR: %w", err)
+				}
+			}
+			return nil
+		}
+
+		sampleRatio := "0.1"
```

Comment:
since the "0.1" is the default, is this needed ? (probably the type should be changed to `string` instead of `*string`

Reply:
since it's default can't i just remove the if check and keep it a `*string`, i don't see a reason for it to be a `string`

Reply:
not sure what value would the *string bring, that would only be useful to discriminate from an zero value and not set, but if there is adefault value, it can't be not set, am I missing something ?